File: backjoon/백트래킹/Python/N과M(3).py
def solve(N, K):
  
    result = []

    def dfs():
        if(len(result) == K) :
            print(' '.join(map(str, result)))
            return
        
        for i in range(1, N+1):
            result.append(i)
            dfs()
            result.pop()

    dfs()

# 메인 함수
if __name__ == "__main__":
    N, K = map(int, input().split())
    solve(N, K)

File: backjoon/백트래킹/Js/N과M(3).js
const input = require("fs")
  .readFileSync(process.platform === "linux" ? "./dev/stdin" : "./input.txt")
  .toString()
  .trim()
  .split("\n");

const [n, m] = input.shift().split(" ").map(Number);
let result = "";

function solution(n, m) {
  let seq = Array(m).fill(0);

  const dfs = (k) => {
    if (k === m) {
      const arr = [];
      for (let i = 0; i < m; i++) {
        arr.push(seq[i]);
      }

      return (result += `${arr.join(" ")}\n`);
    }

    for (let i = 1; i <= n; i++) {
      seq[k] = i;
      dfs(k + 1);
    }
  };

  dfs(0);
  return result;
}

console.log(solution(n, m));
